meshtastic/tests/test_node_runtime_seturl.py

```
@@ -0,0 +1,1498 @@
+        assert result.channel_set.lora_config.hop_limit == 3
+
+    @pytest.mark.unit
+    def test_parse_without_hash_raises_error(self) -> None:
+        """parse() with URL missing hash fragment raises error."""
+        url = "https://meshtastic.org/d/"
+
+        captured_msg: list[str] = []
+
+        def raise_error(msg: str) -> NoReturn:
+            captured_msg.append(msg)
+            raise Exception(msg)
```

Ruff (0.15.6)

[warning] Create your own exception

(TRY002)